migrations/000048_HealthAuthorityStats.up.sql

```diff
@@ -0,0 +1,71 @@
+-- Copyright 2020 Google LLC
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+BEGIN;
+
+CREATE TABLE HealthAuthorityStatsRelease(
+    health_authority_id INT NOT NULL REFERENCES HealthAuthority(id),
```

Comment:
AFAIK, we aren't actively using any FK references for our other tables. There's implications to doing this, typically around deleting and cascading deletes.

Reply:
there are several FK relationships in the DB already

```
❯ grep -i references migrations/*up*
migrations/000001_initial.up.sql:	query_id VARCHAR(50) NOT NULL REFERENCES FederationQuery (query_id),
migrations/000001_initial.up.sql:	sync_id INT REFERENCES FederationSync (sync_id)
migrations/000001_initial.up.sql:	config_id INT NOT NULL REFERENCES ExportConfig(config_id),
migrations/000001_initial.up.sql:	batch_id INT REFERENCES ExportBatch(batch_id),
migrations/000026_health_authority_config.up.sql:	health_authority_id INT NOT NULL REFERENCES HealthAuthority(id),
migrations/000031_Addv1dot5Columns.up.sql:	ADD COLUMN health_authority_id INT REFERENCES HealthAuthority(id),
migrations/000038_add_imports.up.sql:    export_import_id INT NOT NULL REFERENCES ExportImport(id),
migrations/000038_add_imports.up.sql:    export_import_id INT NOT NULL REFERENCES ExportImport(id),
migrations/000042_ReAddExportImporterTables.up.sql:    export_import_id INT NOT NULL REFERENCES ExportImport(id),
migrations/000042_ReAddExportImporterTables.up.sql:    export_import_id INT NOT NULL REFERENCES ExportImport(id),
migrations/000044_AddMirrorSettings.up.sql:    mirror_id INT REFERENCES Mirror(id),
migrations/000048_HealthAuthorityStats.up.sql:    health_authority_id INT NOT NULL REFERENCES HealthAuthority(id),
migrations/000048_HealthAuthorityStats.up.sql:    health_authority_id INT NOT NULL REFERENCES HealthAuthority(id),
```

Reply:
We should consider adding a `ON DELETE` here. It could be:

- `CASCADE` - delete these stats too if the HA is deleted
- `RESTRICT` - don't allow the HA to be deleted if stats exist